processing/src/main/java/io/druid/query/aggregation/DoubleMaxAggregatorFactory.java

```diff
@@ -52,25 +55,46 @@ public DoubleMaxAggregatorFactory(String name, String fieldName)
   }
 
   @Override
-  public Aggregator factorize(ColumnSelectorFactory metricFactory)
+  public Pair<Aggregator, BaseNullableColumnValueSelector> factorize2(ColumnSelectorFactory metricFactory)
   {
-    return new DoubleMaxAggregator(getDoubleColumnSelector(metricFactory, Double.NEGATIVE_INFINITY));
+    BaseDoubleColumnValueSelector doubleColumnSelector = getDoubleColumnSelector(
+        metricFactory,
+        Double.NEGATIVE_INFINITY
+    );
+    return Pair.of(
```

Comment:
Instead of doing this, maybe we add method `getSelector()` to `Aggregator`, and return just aggregator from `factorize2()`? It could throw `UnsupportedOperationException` when it doesn't apply.